magento/magento2
# 11882: It's not possible to enable "log to file" (debugging) in production mode. Psr logger debug method does not work by the default in developer mode.

### Description
Adds ability to enable "log to file" in production mode. Enable by default "log to file" in developer mode.

### Fixed Issues (if relevant)

1. magento/magento2#11882: It's not possible to enable "log to file" (debugging) in production mode
2. magento/magento2#11509: Psr logger debug method does not work by the default in developer mode

### Manual testing scenarios
1. Install Magento.
2. Switch to developer mode: bin/magento deploy:mode:set developer.
3. Login to admin.
4. Navigate to Stores > Configuration > Advanced > Developer > Debug.
5. Check "Log to File" set to Yes.
6. Switch to production mode: bin/magento deploy:mode:set production.
7. Navigate to Stores > Configuration > Advanced > Developer > Debug.
8. Check "Log to File" set to No. You have ability to change "Log to File" to Yes.

### Contribution checklist
